### PR TITLE
Add an example node that calls a .NET method at runtime

### PR DESCRIPTION
**Pull request by gregstoll**
[x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nidevlabs/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

This adds an example node that calls .NET code at runtime.

### Why should this Pull Request be merged?

Partners have been interested in doing this, and the .NET Interface document is currently insufficient for their needs.

### What testing has been done?

Built the project and ran NXG - verified that the node appears in the palette, looks correct, and acts correctly at runtime.

### CODE REVIEW

File: ExamplePlugins/ExampleNode/Model/CalculateTotalLengthCallback.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ExamplePlugins.ExampleNode.Model
+{
+    /// <summary>
+    /// An example class that is called into by <see cref="CalculateTotalLengthDfirNode"/>
```
Comment (ccifra) [inline ExamplePlugins/ExampleNode/Model/CalculateTotalLengthCallback.cs]: I would be a little clearer that the DFIR node generates code that will call into this method at runtime by the execution system.

Comment (ccifra) [inline ExamplePlugins/ExampleNode/Model/CalculateTotalLengthCallback.cs]: I would be a little clearer that the DFIR node generates code that will call into this method at runtime by the execution system.

File: ExamplePlugins/ExampleNode/Model/CalculateTotalLengthNode.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        {
+            get
+            {
+                yield return _inputNamesTerminal;
+                yield return _inputExtraNameTerminal;
+                yield return _outputTerminal;
+            }
+        }
+
+        /// <inheritdoc />
```
Comment (ccifra) [inline ExamplePlugins/ExampleNode/Model/CalculateTotalLengthNode.cs]: It is best not to use inheritdoc in the examples.

**Review by ccifra — CHANGES_REQUESTED** [review]
Can we tighten up the documentation on the examples?

File: ExamplePlugins/ExampleNode/Model/CalculateTotalLengthNode.cs (hunk 2 above)
Comment (gregstoll) [reply ExamplePlugins/ExampleNode/Model/CalculateTotalLengthNode.cs]: Got rid of inheritdoc's in this node and the other example node.

**Comment by gregstoll** [comment]
Addressed the comments - let me know what you think.  (not sure if it's safe to remove and readd you on the review in GitHub :-) )

**Pull request merged**